WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormTableViewDataSource.swift

```diff
@@ -53,13 +53,36 @@ private extension ProductFormTableViewDataSource {
 private extension ProductFormTableViewDataSource {
     func configureCellInPrimaryFieldsSection(_ cell: UITableViewCell, row: ProductFormSection.PrimaryFieldRow) {
         switch row {
+        case .name(let name):
+            configureName(cell: cell, name: name)
         case .description(let description):
             configureDescription(cell: cell, description: description)
         default:
             fatalError("Not implemented yet")
         }
     }
 
+    func configureName(cell: UITableViewCell, name: String?) {
```

Comment:
Would it be possible to create two different name rows in the enum? Maybe a row called `.noName` and a row called `.name`?

That way you could check the string much earlier in the code and can switch on a row `where row == .noName` and it could call a function `configureName(cell:)`. If I understand what you want to do here, it would eliminate casting the cells and the checks for the name string.

Reply:
I started this for Product description (the case below), and here are my thoughts:
- which cell type to display for the row is an implementation detail on the cell configuration side (which is handled by this `ProductFormTableViewDataSource`). Using different enums for different row states sort of leaks this detail to whoever sets up the sections and rows (`ProductFormTableViewModel` in this case). If one day we decide to just use one cell for both empty and non-empty Product name/description, we'd have to also update the enum and `ProductFormTableViewModel`.
- some form of manual typecasting is necessary for each cell configuration unless we have a generics way of setting up the table view data. I remember @ctarda wrote a post about it (p99K0U-1kn-p2) and it'd be great for the team to adopt a table view setup pattern that is less boilerplate, more generics and type-safe 😃

what do you think @pmusolino about one row enum vs. two row enums for two cell states based on the data? also, feel free to propose different ways of setting up table view if you see anything confusing using the current setup

Reply:
Looking at what you wrote, I've not considered switching to the two-row enums, just because I think in the future it will become massive, like the Order Detail Datasource. Looking at the design of this view, in the future, we will have a lot of states based on the data for the cells. The risk is to have two or more row enums for every single row, only to manage the state of a cell. Does it seem that this is the first point in the application where this need appears, or am I wrong?

Reply:
I don't have the deep-level knowledge that you two have with products, so I trust you're doing what's appropriate for this feature. What I observe is that if/else + guard statements inside of each looks like a code smell. I don't have the finer points of implementation in my head, so I can't give any constructive suggestions on ways to make it better. 🤷‍♀